### PR TITLE
Upgrade react router and styled-components

### PR DESCRIPTION
**Pull request by taylrj**
## Change list
- Upgrade react router@^4.3.1
- Upgrade styled-components@^4.1.3
- Release npm package `6.1.0-bata.4`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,7 +58,8 @@
     "lodash": "^4.17.4",
     "prop-types": "^15.5.10",
     "react-ga": "^2.5.3",
-    "react-router": "^3.0.0",
+    "react-router": "^4.3.1",
+    "react-router-dom": "^4.3.1",
```
Comment (nickhsine) [inline package.json]: update `react-router` in the `peerDependencies`.

File: package.json [hunk 2]
```diff
@@ -58,7 +58,8 @@
     "lodash": "^4.17.4",
     "prop-types": "^15.5.10",
     "react-ga": "^2.5.3",
-    "react-router": "^3.0.0",
+    "react-router": "^4.3.1",
```
Comment (nickhsine) [inline package.json]: `react-router-dom` dependents on `react-router`, so we can  skip this line.

File: error/src/components/error-message.js [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
 import Eng404 from '../../static/not-found-eng.svg'
 import Eng404Mobile from '../../static/not-found-eng-mobile.svg'
+import Eng500 from '../../static/server-error.svg'
+import Eng500Mobile from '../../static/server-error-mobile.svg'
 import Number404 from '../../static/num404.svg'
 import Number500 from '../../static/num500.svg'
 import PropTypes from 'prop-types'
 import React from 'react'
 import styled from 'styled-components'
+import { Link } from 'react-router-dom'
```
Comment (nickhsine) [inline error/src/components/error-message.js]: `import Link from 'react-router-dom/Link'` ?

File: package.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "name": "@twreporter/react-components",
   "repository": "https://github.com/twreporter/twreporter-react-components.git",
   "license": "AGPL-3.0",
-  "version": "5.1.3",
+  "version": "6.1.0-beta.1",
```
[the remark this answers is not in the document]
Comment (taylrj) [reply package.json]: It seems that `6.0.0-beta.1`[ has been released](https://www.npmjs.com/package/@twreporter/react-components) before.

**Comment by taylrj** [comment]
Updated. And release a new beta version `6.1.0-beta.2` to npm package.

**Comment by nickhsine** [comment]
Before merging this PR, please check this migration doc(https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/guides/migrating.md#link).
 `to` prop needs to be specified in `react-router@4`, if not, there will be an error.

**Comment by taylrj** [comment]
Updated and released a new beta version `6.1.0-beta.3`

**Comment by nickhsine** [comment]
LGTM

**Comment by taylrj** [comment]
Updated. 
- Upgrade styled-components@^4.1.3 
- Release npm package `6.1.0-bata.4`

File: package.json [hunk 5]
```diff
@@ -42,13 +42,13 @@
...
+    "react-dom": "^16.3.0",
     "rimraf": "^2.6.1"
   },
   "peerDependencies": {
-    "react": "^15.4.1 || ^16.0.0",
-    "react-dom": "^15.4.1 || ^16.0.0",
+    "react": "^16.3.0",
+    "react-dom": "^16.3.0",
     "react-helmet": "^4.0.0 || ^5.0.0",
     "react-router": "^4.3.1"
```
Comment (nickhsine) [inline package.json]: could we change to `"react-router": "^4.0.0"` ?

File: package.json (hunk 1 above)
Comment (nickhsine) [reply package.json]: Could we change to `react-router-dom: ^4.0.0` ?

File: package.json [hunk 6]
```diff
@@ -42,27 +42,27 @@
...
     "prop-types": "^15.5.10",
     "react-ga": "^2.5.3",
-    "react-router": "^3.0.0",
+    "react-router-dom": "^4.3.1",
     "react-swipeable": "^4.0.1",
     "react-transition-group": "^1.2.1",
     "react-waypoint": "^7.0.3",
     "smoothscroll": "^0.3.0",
-    "styled-components": "^3.0.0"
+    "styled-components": "^4.1.3"
```
Comment (nickhsine) [inline package.json]: Could we change to `styled-components@^4.0.0`

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,6 @@
 ### UNRELEASED
-- Upgrade react-router to v4.3.1
+- Upgrade react-router@^4.3.1
```
Comment (nickhsine) [inline CHANGELOG.md]: `Upgrade react-router-dom@^4.3.1` ?

**Comment by taylrj** [comment]
Updated.

Release npm package `6.1.0-bata.5`

**Pull request merged**